### PR TITLE
Makes the Light Replacer more easily attainable + Allows broken light recycling

### PR DESCRIPTION
**Pull request by TheKillfish**
# About the pull request

Adds the Light Replacer to both ComTech and normal tool storage vendors, and adds it to the autolathe recipe list. Allows the Light Replacer to recycle broken lights either when replacing bulbs or inserting a broken one by hand, converting 5 broken ones to 1 usable light.

Also cleaned up a grammar mistake when shattering a lightbulb.

# Explain why it's good for the game

The Light Replacer is a niche but handy tool for lighting places a bit more effectively than flares or flashlights can (provided the local APC is still functioning or that there even is one) but has very limited avaliability. Though not a major issue, it'd be nicer to have a way of replacing/aquiring one than having to steal one of the limited few from Upper Engineering, hunting for one groundside, or bugging your local engineers for theirs if they aren't using it (most of whom probably ditched it in their prep area anyway). Plus, it'd make sense that tool vendors would have them in stock and that the autolathe can print replacements.

The minor rework improves functionality and makes using it a smoother process.

Bad grammar is bad.

# Testing Photographs and Procedure
Added it to autolathe recipes and vendors. Found out it didn't have material costs, so added matter values.

# Changelog

:cl: Killfish
add: Added the Light Replacer to tool storage and to the autolathe. The Light Replacer can now also recycle broken bulbs, either directly when replacing or when inserted by hand.
spellcheck: Fixes missing 'the' when shattering a light bulb, so it no longer says 'light tube smashes' in all lower case.
/:cl:

### CODE REVIEW

**Comment by TheKillfish** [comment]
Blundir suggested broken light recycling, so I've done that now.

File: code/modules/power/lighting.dm [hunk 1]
```diff
@@ -758,7 +758,7 @@
 
 /obj/item/light_bulb/proc/shatter()
 	if(status == LIGHT_OK || status == LIGHT_BURNED)
-		src.visible_message(SPAN_DANGER("[name] shatters."),SPAN_DANGER("You hear a small glass object shatter."))
+		src.visible_message(SPAN_DANGER("The [name] shatters."),SPAN_DANGER("You hear a small glass object shatter."))
```
Comment (harryob) [inline code/modules/power/lighting.dm:761]:
```suggestion
		visible_message(SPAN_DANGER("[src] shatters."), SPAN_DANGER("You hear a small glass object shatter."))
```

File: code/game/objects/items/devices/lightreplacer.dm [hunk 2]
```diff
@@ -116,6 +127,17 @@
 		AddUses(1)
 		charge = 1
 
+/obj/item/device/lightreplacer/proc/Recycle(mob/living/U)
+	if(recycle == max_recycle)
+		recycle = 0
+		AddUses(1)
+		playsound(src.loc, 'sound/machines/ding.ogg', 5, 1)
```
Comment (harryob) [inline code/game/objects/items/devices/lightreplacer.dm:134]:
```suggestion
		playsound(src, 'sound/machines/ding.ogg', 5, 1)
```

File: code/game/objects/items/devices/lightreplacer.dm [hunk 3]
```diff
@@ -116,6 +127,17 @@
...
+		return
+	else
+		playsound(src.loc, 'sound/machines/click.ogg', 25, 1)
```
Comment (harryob) [inline code/game/objects/items/devices/lightreplacer.dm:137]:
```suggestion
		playsound(src, 'sound/machines/click.ogg', 25, 1)
```

File: code/game/objects/items/devices/lightreplacer.dm [hunk 4]
```diff
@@ -83,31 +91,34 @@
...
 
 /obj/item/device/lightreplacer/proc/Use(mob/user)
 
-	playsound(src.loc, 'sound/machines/click.ogg', 25, 1)
 	AddUses(-1)
 	return 1
 
 // Negative numbers will subtract
 /obj/item/device/lightreplacer/proc/AddUses(amount = 1)
+	playsound(src.loc, 'sound/machines/click.ogg', 25, 1)
```
Comment (harryob) [inline code/game/objects/items/devices/lightreplacer.dm:121]:
```suggestion
	playsound(src, 'sound/machines/click.ogg', 25, 1)
```

File: code/game/objects/items/devices/lightreplacer.dm [hunk 5]
```diff
@@ -83,31 +91,34 @@
...
 				AddUses(1)
-				to_chat(user, "You insert the [L.name] into the [src.name]. You have [uses] lights remaining.")
+				to_chat(user, SPAN_NOTICE("You insert the [L.name] into the [src.name]. You have [uses] lights remaining."))
 				user.drop_held_item()
 				qdel(L)
 				return
 		else
-			to_chat(user, "You need a working light.")
+			Recycle()
+			to_chat(user, SPAN_NOTICE("You insert the [L.name] into the [src.name] for recycling."))
```
Comment (harryob) [inline code/game/objects/items/devices/lightreplacer.dm:100]:
```suggestion
			to_chat(user, SPAN_NOTICE("You insert [L] into [src] for recycling."))
```

File: code/game/objects/items/devices/lightreplacer.dm [hunk 6]
```diff
@@ -83,31 +91,34 @@
 		if(L.status == 0) // LIGHT OKAY
 			if(uses < max_uses)
 				AddUses(1)
-				to_chat(user, "You insert the [L.name] into the [src.name]. You have [uses] lights remaining.")
+				to_chat(user, SPAN_NOTICE("You insert the [L.name] into the [src.name]. You have [uses] lights remaining."))
```
Comment (harryob) [inline code/game/objects/items/devices/lightreplacer.dm:94]:
```suggestion
				to_chat(user, SPAN_NOTICE("You insert [L] into [src]. You have [uses] lights remaining."))
```

File: code/game/objects/items/devices/lightreplacer.dm [hunk 7]
```diff
@@ -116,6 +127,17 @@
...
+/obj/item/device/lightreplacer/proc/Recycle(mob/living/U)
+	if(recycle == max_recycle)
+		recycle = 0
+		AddUses(1)
+		playsound(src.loc, 'sound/machines/ding.ogg', 5, 1)
+		return
+	else
+		playsound(src.loc, 'sound/machines/click.ogg', 25, 1)
+		recycle += 1
+		return
```
Comment (harryob) [inline code/game/objects/items/devices/lightreplacer.dm:139]:
```suggestion

	playsound(src.loc, 'sound/machines/click.ogg', 25, 1)
	recycle += 1
```
the else clause is redundant as the previous condition returns

**Review by harryob — CHANGES_REQUESTED** [review]

**Pull request closed without merging**